scripts/train_darcy_transformer.py

```diff
@@ -0,0 +1,210 @@
+import sys
+
+from configmypy import ConfigPipeline, YamlConfig, ArgparseConfig
+import torch
+from torch.nn.parallel import DistributedDataParallel as DDP
+import wandb
+
+from neuralop import H1Loss, LpLoss, Trainer, get_model
+from neuralop.datasets import load_darcy_flow_small
+from neuralop.training import setup
+from neuralop.training.callbacks import MGPatchingCallback, SimpleWandBLoggerCallback
+from neuralop.utils import get_wandb_api_key, count_model_params
+
+
+# Read the configuration
+config_name = "default"
+pipe = ConfigPipeline(
+    [
+        YamlConfig(
+            "./darcy_transformer_config.yaml", config_name="default", config_folder="../config"
+        ),
+        ArgparseConfig(infer_types=True, config_name=None, config_file=None),
+        YamlConfig(config_folder="../config"),
+    ]
+)
+config = pipe.read_conf()
+config_name = pipe.steps[-1].config_name
+
+# Set-up distributed communication, if using
+device, is_logger = setup(config)
+
+# Set up WandB logging
+if config.wandb.log and is_logger:
+    wandb.login(key=get_wandb_api_key())
+    if config.wandb.name:
+        wandb_name = config.wandb.name
+    else:
+        wandb_name = "_".join(
+            f"{var}"
+            for var in [
+                config_name,
+                config.transformer_no.encoder_n_layers,
+                config.transformer_no.encoder_hidden_channels,
+                config.transformer_no.decoder_hidden_channels,
+                config.transformer_no.encoder_num_heads,
+                config.transformer_no.decoder_num_heads,
+                config.patching.levels,
+                config.patching.padding,
+            ]
+        )
+    wandb_args =  dict(
+        config=config,
+        name=wandb_name,
+        group=config.wandb.group,
+        project=config.wandb.project,
+        entity=config.wandb.entity,
+    )
+    if config.wandb.sweep:
+        for key in wandb.config.keys():
+            config.params[key] = wandb.config[key]
+
+# Make sure we only print information when needed
+config.verbose = config.verbose and is_logger
+
+# Print config to screen
+if config.verbose and is_logger:
+    pipe.log()
+    sys.stdout.flush()
+
+# Loading the Darcy flow dataset
+train_loader, test_loaders, output_encoder = load_darcy_flow_small(
+    n_train=config.data.n_train,
+    batch_size=config.data.batch_size,
+    positional_encoding=config.data.positional_encoding,
+    test_resolutions=config.data.test_resolutions,
+    n_tests=config.data.n_tests,
+    test_batch_sizes=config.data.test_batch_sizes,
+    encode_input=config.data.encode_input,
+    encode_output=config.data.encode_output,
+)
+
+model = get_model(config)
+
+### The following wrapper is to maintain a consistent input format with other FNO-based model
```

Comment:
Is there a reason why we do not directly use the FNO input format for the `TransformerNO`?

Reply:
@dhpitt seems like we'd want to use the refactored 2DGrid here instead.